jbmc/src/java_bytecode/java_utils.cpp

```diff
@@ -285,7 +285,8 @@ exprt make_function_application(
   declare_function(fun_name, type, symbol_table);
 
   // Function application
-  return function_application_exprt(symbol_exprt(fun_name), arguments, type);
+  return function_application_exprt(
+    symbol_exprt(fun_name, typet()), arguments, type);
```

Comment:
This function doesn't seem correct. I think it should be changed to take a `code_typet` as argument representing the type of the function instead of `type` (the return type can be deduced from the `code_typet`